FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -18,7 +18,49 @@ public final class RemoteFeedLoader: FeedLoader {
 		self.client = client
 	}
 
-	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+	public func load(completion: @escaping (RemoteFeedLoader.Result) -> Void) {
+		client.get(from: url, completion: { [weak self] result in
+
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, response)):
+				completion(FeedImageMapper.map(data, response))
+			case .failure:
+				completion(.failure(Error.connectivity))
+			}
+		})
+	}
+}
+
+private final class FeedImageMapper {
+	private struct Root: Decodable {
+		var items: [Item]
+	}
+
+	private struct Item: Decodable {
+		let image_id: UUID
+		let image_desc: String?
+		let image_loc: String?
+		let image_url: URL
+
+		var item: FeedImage {
+			return FeedImage(id: image_id,
+			                 description: image_desc,
+			                 location: image_loc,
+			                 url: image_url)
+		}
+	}
+
+	private static var OK_200: Int { return 200 }
+
+	static func map(_ data: Data, _ response: HTTPURLResponse) -> RemoteFeedLoader.Result {
+		guard response.statusCode == OK_200,
+		      let root = try? JSONDecoder().decode(Root.self, from: data) else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+
+		let items = root.items.map { $0.item }
```

Comment:
Suggestion: another way of achieving the same result with `map` is to make use of key path:
```suggestion
		let items = root.items.map(\.item)
```